federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -65,10 +67,12 @@ func NewclusterController(federationClient federationclientset.Interface, cluste
 	cc.clusterStore.Store, cc.clusterController = cache.NewInformer(
 		&cache.ListWatch{
 			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
```

Comment:
The next step now is to write a VersionedListWatch which will take v1.ListOptions as argument instead of the unversioned one. That way these controllers will not need to depend on `pkg/api` at all, which will be awesome!